src/software/ai/hl/stp/play/kickoff_play_test.py

```diff
@@ -2,92 +2,144 @@
-            tbots_cpp.Point(-3, -2.5),
-        ]
-
-        yellow_bots = [
-            tbots_cpp.Point(1, 0),
-            tbots_cpp.Point(1, 2.5),
-            tbots_cpp.Point(1, -2.5),
-            tbots_cpp.Field.createSSLDivisionBField().enemyGoalCenter(),
-            tbots_cpp.Field.createSSLDivisionBField()
-            .enemyDefenseArea()
-            .negXNegYCorner(),
-            tbots_cpp.Field.createSSLDivisionBField()
-            .enemyDefenseArea()
-            .negXPosYCorner(),
-        ]
-
-        blue_play = Play()
-        yellow_play = Play()
-
-        # Game Controller Setup
+    # starting point must be Point
+    ball_initial_pos = tbots.Point(0, 0)
+
+    # Setup Bots
+    blue_bots = [
+        tbots.Point(-3, 2.5),
+        tbots.Point(-3, 1.5),
+        tbots.Point(-3, 0.5),
+        tbots.Point(-3, -0.5),
+        tbots.Point(-3, -1.5),
+        tbots.Point(-3, -2.5),
+    ]
+
+    yellow_bots = [
+        tbots.Point(1, 0),
+        tbots.Point(1, 2.5),
+        tbots.Point(1, -2.5),
+        tbots.Field.createSSLDivisionBField().enemyGoalCenter(),
+        tbots.Field.createSSLDivisionBField().enemyDefenseArea().negXNegYCorner(),
+        tbots.Field.createSSLDivisionBField().enemyDefenseArea().negXPosYCorner(),
+    ]
+
+    blue_play = Play()
+    yellow_play = Play()
+
+    # Game Controller Setup
+    simulated_test_runner.gamecontroller.send_gc_command(
+        gc_command=Command.Type.STOP, team=Team.UNKNOWN
+    )
+
+    if is_friendly_test:
         simulated_test_runner.gamecontroller.send_gc_command(
-            gc_command=Command.Type.STOP, team=Team.UNKNOWN
+            gc_command=Command.Type.KICKOFF, team=Team.BLUE
         )
+        blue_play.name = PlayName.KickoffFriendlyPlay
+        yellow_play.name = PlayName.KickoffEnemyPlay
+    else:
         simulated_test_runner.gamecontroller.send_gc_command(
-            gc_command=Command.Type.NORMAL_START, team=Team.BLUE
+            gc_command=Command.Type.KICKOFF, team=Team.YELLOW
+        )
+        blue_play.name = PlayName.KickoffEnemyPlay
+        yellow_play.name = PlayName.KickoffFriendlyPlay
+
+    simulated_test_runner.gamecontroller.send_gc_command(
+        gc_command=Command.Type.NORMAL_START, team=Team.BLUE
+    )
+
+    # Force play override here
+    simulated_test_runner.blue_full_system_proto_unix_io.send_proto(Play, blue_play)
+    simulated_test_runner.yellow_full_system_proto_unix_io.send_proto(Play, yellow_play)
+
+    # Create world state
+    simulated_test_runner.simulator_proto_unix_io.send_proto(
+        WorldState,
+        create_world_state(
+            yellow_robot_locations=yellow_bots,
+            blue_robot_locations=blue_bots,
+            ball_location=ball_initial_pos,
+            ball_velocity=tbots.Vector(0, 0),
+        ),
+    )
+
+    # Always Validation: Check that robots are within centreCircle + friendlyHalf
+    always_validation_sequence_set = [[]]
+
+    if is_friendly_test:
```

Comment:
this if statement could probably be condensed a bit, so this can be common between both branches:

```
always_validation_sequence_set[0].append(
            NumberOfRobotsAlwaysStaysInRegion(
                regions=[
                    tbots.Field.createSSLDivisionBField().friendlyHalf(),
                    tbots.Field.createSSLDivisionBField().centerCircle(),
                ],
                req_robot_cnt=6,
            )
        )
always_validation_sequence_set[0].append(
            OrValidation(
                [
                    NumberOfRobotsAlwaysStaysInRegion(
                        regions=[tbots.Field.createSSLDivisionBField().centerCircle()],
                        req_robot_cnt=0,
                    ),
                ]
            )
```

and you can just append this to the OR validation in the friendly case
```
 NumberOfRobotsAlwaysStaysInRegion(
           regions=[tbots.Field.createSSLDivisionBField().centerCircle()],
           req_robot_cnt=1,
),
```

Reply:
Done